manifests/client/host_config_entry.pp

```diff
@@ -377,10 +381,376 @@
+      value => ssh::config_bool_translate($identitiesonly),
+    ;
+    "${_name}__KbdInteractiveAuthentication":
+      key   => 'KbdInteractiveAuthentication',
+      value => ssh::config_bool_translate($kbdinteractiveauthentication),
+    ;
+    "${_name}__LogLevel":
+      key   => 'LogLevel',
+      value => $ssh_loglevel,
+    ;
+    "${_name}__MACs":
+      key   => 'MACs',
+      value => $_macs,
+    ;
+    "${_name}__NoHostAuthenticationForLocalhost":
+      key   => 'NoHostAuthenticationForLocalhost',
+      value => ssh::config_bool_translate($nohostauthenticationforlocalhost),
+    ;
+    "${_name}__NumberOfPasswordPrompts":
+      key   => 'NumberOfPasswordPrompts',
+      value => String($numberofpasswordprompts),
+    ;
+    "${_name}__PasswordAuthentication":
+      key   => 'PasswordAuthentication',
+      value => ssh::config_bool_translate($passwordauthentication),
+    ;
+    "${_name}__PermitLocalCommand":
+      key   => 'PermitLocalCommand',
+      value => ssh::config_bool_translate($permitlocalcommand),
+    ;
+    "${_name}__Port":
+      key   => 'Port',
+      value => String($port),
+    ;
+    "${_name}__PreferredAuthentications":
+      key   => 'PreferredAuthentications',
+      value => $preferredauthentications.join(','),
+    ;
+    "${_name}__PubkeyAuthentication":
+      key   => 'PubkeyAuthentication',
+      value => ssh::config_bool_translate($pubkeyauthentication),
+    ;
+    "${_name}__RhostsRSAAuthentication":
+      key   => 'RhostsRSAAuthentication',
+      value => ssh::config_bool_translate($rhostsrsaauthentication),
+    ;
+    "${_name}__RSAAuthentication":
+      key   => 'RSAAuthentication',
+      value => ssh::config_bool_translate($rsaauthentication),
+    ;
+    "${_name}__SendEnv":
+      key   => 'SendEnv',
+      value => $sendenv,
+    ;
+    "${_name}__ServerAliveCountMax":
+      key   => 'ServerAliveCountMax',
+      value => String($serveralivecountmax),
+    ;
+    "${_name}__ServerAliveInterval":
+      key   => 'ServerAliveInterval',
+      value => String($serveraliveinterval),
+    ;
+    "${_name}__StrictHostKeyChecking":
+      key   => 'StrictHostKeyChecking',
+      value => $stricthostkeychecking,
+    ;
+    "${_name}__TCPKeepAlive":
+      key   => 'TCPKeepAlive',
+      value => ssh::config_bool_translate($tcpkeepalive),
+    ;
+    "${_name}__Tunnel":
+      key   => 'Tunnel',
+      value => $tunnel,
+    ;
+    "${_name}__UsePrivilegedPort":
+      key   => 'UsePrivilegedPort',
+      value => ssh::config_bool_translate($useprivilegedport),
+    ;
+    "${_name}__VerifyHostKeyDNS":
+      key   => 'VerifyHostKeyDNS',
+      value => $verifyhostkeydns,
+    ;
+    "${_name}__VisualHostKey":
+      key   => 'VisualHostKey',
+      value => ssh::config_bool_translate($visualhostkey),
+    ;
+    "${_name}__XAuthLocation":
+      key   => 'XAuthLocation',
+      value => $xauthlocation,
+    ;
+  }
+
+  if $_cipher {
+    ssh_config{ "${_name}__Cipher":
```

Comment:
These could also be collapsed into the same augeas call with a little bit of fancy footwork.